src/main/java/org/cloudfoundry/promregator/config/OAuth2XSUAAAuthenticationConfiguration.java

```diff
@@ -1,45 +1,114 @@
 package org.cloudfoundry.promregator.config;
 
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.http.client.utils.URIBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class OAuth2XSUAAAuthenticationConfiguration {
-	private String tokenServiceURL;
-	
+
+	public final static String deprecatedTokenServciceURLProperty = "tokenServiceURL";
+	public final static String useInsteadXsuassServiceURLProperty = "xsuaaServiceURL";
+
+	private static final Logger log = LoggerFactory.getLogger(OAuth2XSUAAAuthenticationConfiguration.class);
+
+	private String xsuaaServiceURL;
+
+	private String xsuaaServiceCertURL;
+
 	private String client_id;
-	
+
 	private String client_secret;
 
-	private String scopes;
+	private String client_certificates;
+
+	private String client_key;
 
-	public String getTokenServiceURL() {
-		return tokenServiceURL;
+	private final Set<String> scopes = new HashSet<>();
+
+	public String getXsuaaServiceURL() {
+		return xsuaaServiceURL;
 	}
 
-	public void setTokenServiceURL(String tokenServiceURL) {
-		this.tokenServiceURL = tokenServiceURL;
+	public void setTokenServiceURL(String url) {
+
+		String urlSuffix = "/oauth/token";
```

Comment:
Make this a constant `private static final`.
Perhaps also consider renaming it something like `DEEFAULT_OAUTH_TOKEN_URL_PATH` or so.

Reply:
done.